tools/deployJournals.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * @license
+ * Copyright 2022 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+const path_     = require('path');
+const fs_       = require('fs');
+const path = path_.posix;
+
+require('../src/foam_node.js');
+
+var [argv, X, flags] = require('./processArgs.js') (
+  '',
+  { version: '', license: '', pom: 'pom' },
+  { debug: true, java: false, web: true }
+);
+
+foam.require(X.pom, false, true);
+
+const outPath = 'runtime/journals';
+
+if ( ! fs_.existsSync(outPath) ) {
+    fs_.mkdirSync(outPath, { recursive: true });
+}
+
+async function findJournals({ jrls, srcPath, jrlName }) {
+    const walker = foam.util.filesystem.FileWalker.create();
+    walker.files.sub((_1, _2, info) => {
+        for ( const fileInfo of info.files ) {
+            if ( fileInfo.path !== srcPath &&
+              foam.poms[foam.poms.findIndex(p => p.location == fileInfo.path)] != undefined ) {
+              walker.skip.pub();
+              return;
+            }
+            const extName = path_.extname(fileInfo.name);
+            if ( extName !== '.jrl' ) continue;
+            if ( jrlName && fileInfo.name != jrlName ) continue;
```

Comment:
jrlName is always undefined now - maybe this condition can be removed to make the code easier to understand